crates/sui-adapter/src/adapter.rs

```diff
@@ -900,20 +900,21 @@ pub fn resolve_and_type_check(
                 t @ SignatureToken::Struct(_)
                 | t @ SignatureToken::StructInstantiation(_, _)
                 | t @ SignatureToken::TypeParameter(_) => {
-                    if !object.is_owned_or_quasi_shared() {
-                        // Forbid passing shared (both mutable and immutable) object by value.
-                        // This ensures that shared object cannot be transferred, deleted or wrapped.
-                        return Err(ExecutionError::new_with_source(
-                            ExecutionErrorKind::entry_argument_error(
-                                idx,
-                                EntryArgumentErrorKind::InvalidObjectByValue,
-                            ),
-                            format!(
-                                "Only owned object can be passed by-value, violation found in \
-                                argument {}",
-                                idx
-                            ),
-                        ));
+                    match &object.owner {
+                        Owner::AddressOwner(_) | Owner::ObjectOwner(_) | Owner::Shared => (),
+                        Owner::Immutable => {
```

Comment:
Our validator core is not ready to support consuming shared object by-value. There is more work need to be done there.

Reply:
Gotcha. I could just comment that bit out for now?